### PR TITLE
Yet More Config Tweaks

### PR DESCRIPTION
**Pull request by cycomachead**
Adding the `resolver` line for DNS somehow makes nginx unhappy on macOS. This fixes that but also cleans up the config.lua file and fixes the macOS install script.

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -1,20 +1,18 @@
-worker_processes ${{NUM_WORKERS}};
-error_log stderr notice;
-daemon ${{USE_DAEMON}};
-pid logs/nginx.pid;
 env PORT;
 env SSL_PORT;
 env SSL_CERT_NAME;
-env DATABASE_USERNAME;
-env DATABASE_URL;
-env DATABASE_PASSWORD;
```
Comment (bromagosa) [inline nginx.conf]: Why are we removing these? :thinking:

Comment (cycomachead) [reply nginx.conf]: Feel free to correct me, but my understanding is that the `env` declaration only applies to what nginx should be reading. There's no need for nginx to read the db configuration b/c that's a handled in lapis. (And that means I should also remove `MAIL_`* ones.)

File: config.lua [hunk 2]
```diff
@@ -34,31 +22,39 @@ config({'development', 'production'}, {
...
     ssl_cert_name = os.getenv('SSL_CERT_NAME') or 'snap-cloud.cs10.org',
-    enable_https = true,
     enable_auto_ssl = 'true', -- lapis needs a string
     secret = os.getenv('SESSION_SECRET_BASE'),
     num_workers = 16,
     code_cache = 'on',
 
-    --- TODO: See if we can turn this on without a big hit
-    measure_performance = false,
+    log_directive = 'logs/error.log warning',
```
Comment (bromagosa) [inline config.lua]: In my local setup this is giving me an error:

`nginx: [emerg] invalid log level "warning" in /home/renard/Code/Lua/snapCloud/nginx.conf.compiled:14`

File: nginx.conf (hunk 1 above)
Comment (bromagosa) [reply nginx.conf]: I've just confirmed you do need to whitelist every env var you want to be able to access from within OpenResty, and that means all of them. I'm reverting just these changes.

File: config.lua (hunk 2 above)
Comment (cycomachead) [reply config.lua]: Bleh, you're right. I fixed this for #44, but forgot to migrate the change back to this branch. It should just be `warn`. :)

**Pull request merged**